karapace/schema_reader.py

```diff
@@ -492,52 +465,20 @@ def _handle_msg_schema(self, key: dict, value: Optional[dict]) -> None:
         # won't interfere with the equality. Note: This means it is possible
         # for the REST API to return data that is formatted differently from
         # what is available in the topic.
-        if schema_type_parsed in [SchemaType.AVRO, SchemaType.JSONSCHEMA]:
-            try:
-                schema_str = json.dumps(json.loads(schema_str), sort_keys=True)
-            except json.JSONDecodeError:
-                LOG.error("Schema is not valid JSON")
-                return
-        elif schema_type_parsed == SchemaType.PROTOBUF:
-            try:
-                parsed_schema = ValidatedTypedSchema.parse(SchemaType.PROTOBUF, schema_str)
-                schema_str = str(parsed_schema)
-            except InvalidSchema:
-                LOG.exception("Schema is not valid ProtoBuf definition")
-                return
-
-        if schema_subject not in self.subjects:
-            LOG.info("Adding first version of subject: %r with no schemas", schema_subject)
-            self.subjects[schema_subject] = {"schemas": {}}
-
-        subjects_schemas = self.subjects[schema_subject]["schemas"]
-
-        with self.id_lock:
-            # dedup schemas to reduce memory pressure
-            schema_str = self._hash_to_schema.setdefault(hash(schema_str), schema_str)
-
-            if schema_version in subjects_schemas:
-                LOG.info("Updating entry subject: %r version: %r id: %r", schema_subject, schema_version, schema_id)
-            else:
-                LOG.info("Adding entry subject: %r version: %r id: %r", schema_subject, schema_version, schema_id)
-
+        try:
             typed_schema = TypedSchema(
                 schema_type=schema_type_parsed,
                 schema_str=schema_str,
             )
-            subjects_schemas[schema_version] = {
-                "schema": typed_schema,
-                "version": schema_version,
-                "id": schema_id,
-                "deleted": schema_deleted,
-            }
-
-            self.schemas[schema_id] = typed_schema
-            self.global_schema_id = max(self.global_schema_id, schema_id)
-            if not schema_deleted:
-                self._set_schema_id_on_subject(subject=schema_subject, schema=typed_schema, schema_id=schema_id)
-            else:
-                self._delete_from_schema_id_on_subject(subject=schema_subject, schema=typed_schema)
+        except InvalidSchema:
```

Comment:
I guess this should still be `(InvalidSchema, JSONDecodeError)`, unless we also do [this](https://github.com/aiven/karapace/pull/507/files#r1091759473)?

Reply:
Done.